src/esp/core/Check.h

```diff
@@ -51,26 +51,48 @@
 namespace esp {
 namespace core {
 
-/* The throwInPython function pointer gets filled during Python bindings
+/* The throwAssertInPython function pointer gets filled during Python bindings
    startup. If it's nullptr, we're in plain C++ code. */
-extern void (*throwInPython)(const char*);
+extern void (*throwAssertInPython)(const char*);
 
-// For use in ESP_CHECK
+/* The throwRuntimeInPython function pointer gets filled during Python bindings
+   startup. If it's nullptr, we're in plain C++ code. */
+extern void (*throwRuntimeInPython)(const char*);
+
+// For use in ESP_FATAL
 [[noreturn]] void throwIfInPythonOtherwiseAbort(const char* message);
 
+// For use in ESP_CHECK
+[[noreturn]] void throwIfInPythonOtherwiseExit(const char* message);
+
 }  // namespace core
 }  // namespace esp
 
 /* A runtime check that must pass, otherwise we consider this a fatal runtime
-error. The program terminates with the supplied error message. */
+error due to programmer error. The program terminates with the supplied error
```

Comment:
Instead of adding these comments at the definitions of ESP_FATAL and ESP_CHECK, can you rewrite the comment at the top of the file? We have to talk about all of these in one place so someone can look in one place to decide which error-handling code is appropriate.

Spoiler: I'm skeptical that there's any real difference between ESP_FATAL and assert, so I expect we'll need some discussion to arrive at a consensus here.

Reply:
I was thinking the benefit of ESP_FATAL was that it would provide python message capability.

Reply:
Let's discuss on Slack or in an eng meeting.